content/posts/2021/reana-0.7.4.md

```diff
@@ -0,0 +1,134 @@
+  - cluster
+---
+
+We are glad to announce the release of REANA 0.7.4, an update
+allowing REANA users to specify a memory limit for Kubernetes compute backend jobs.
+
+## What's new?
+
+### Custom memory limit for Kubernetes compute backend
+
+While executing a workflow there could be some cases when a complex workflow
+job with a high memory consumption exceeds the cluster memory limit. In this case
+the cluster will respond with out of memory (OOM) error and the job will be terminated (OOMKilled).
+To avoid such an error and to make an efficient use of the available resources,
+it is recommended to specify a memory limit per workflow step.
+If a memory limit for a specific step is increased, OOM error will not be thrown and
+the workflow will be executed as expected. Also, a memory limit could be decreased
+in order to increment the chances of jobs being scheduled earlier.
+
+REANA 0.7.4 enables to set a memory limit for a workflow jobs by specifying
+`kubernetes_memory_limit` in the specification of each workflow step.
+Please refer to
+[Kubernetes official documentation](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-memory)
+for expected memory values.
+The specifications respectively can be configured for Serial, Yadage,
+and CWL workflow engines.
+
+### Custom memory limit for Serial workflow
+
+We can set `kubernetes_memory_limit` in every step of workflow specification.
+
+
+```diff
+  ...
+  steps:
+    - name: gendata
+      environment: 'reanahub/reana-env-root6:6.18.04'
+      compute_backend: kubernetes
++     kubernetes_memory_limit: '8Gi'
+      commands:
+      - mkdir -p results && root -b -q 'code/gendata.C(${events},"${data}")'
+    - name: fitdata
+      environment: 'reanahub/reana-env-root6:6.18.04'
+      compute_backend: kubernetes
++     kubernetes_memory_limit: '4Gi'
+      commands:
+      - root -b -q 'code/fitdata.C("${data}","${plot}")'
+```
+
+
+### Custom memory limit for Yadage workflow
+
+We can set `kubernetes_memory_limit` in every stage step under `environment.resources`.
+
+```diff
+  ...
+  stages:
+    - name: gendata
+      dependencies: [init]
+      scheduler:
+        scheduler_type: 'singlestep-stage'
+        parameters:
+          events: {step: init, output: events}
+          gendata: {step: init, output: gendata}
+          outfilename: '{workdir}/data.root'
+        step:
+          process:
+            process_type: 'interpolated-script-cmd'
+            script: root -b -q '{gendata}({events},"{outfilename}")'
+          publisher:
+            publisher_type: 'frompar-pub'
+            outputmap:
+              data: outfilename
+          environment:
+            environment_type: 'docker-encapsulated'
+            image: 'reanahub/reana-env-root6'
+            imagetag: '6.18.04'
+            resources:
+              - compute_backend: kubernetes
++             - kubernetes_memory_limit: '8Gi'
+```
+### Custom memory limit for CWL workflow
```

Comment:
Example of custom memory limits for CWL workflows

The `kubernetes_memory_limit` values for CWL jobs can be specified as a workflow hint under the `hints` clause: